webpush/encrypt.go

```diff
@@ -241,11 +294,42 @@ func newInfo(infoType string, context []byte) []byte {
 	info = append(info, []byte("Content-Encoding: ")...)
 	info = append(info, []byte(infoType)...)
 	info = append(info, 0)
-	info = append(info, []byte("P-256")...)
-	info = append(info, context...)
+	// For aes128gcm ctx is not needed.
+	if len(context) > 0 {
+		info = append(info, []byte("P-256")...)
+		info = append(info, context...)
+	}
 	return info
 }
 
+// Returns a keying material. See sections of
+// https://tools.ietf.org/html/draft-ietf-webpush-encryption-07#section-3.3
+func newKeyInfo(subscriptionPublicKey, serverPublicKey []byte) []byte {
+	var info []byte
+	info = append(info, keyInfoPrefix...)
+	info = append(info, subscriptionPublicKey...)
+	info = append(info, serverPublicKey...)
+	return info
```

Comment:
Because you know all the lengths, you could also make it a bit more performant:

```go
info := make([]byte, len(keyInfoPrefix) + len(subscriptionPublicKey) + len(serverPublicKey))
n := copy(info, keyInfoPrefix)
n += copy(info[n:], subscriptionPublicKey)
copy(info[n:], serverPublicKey)
```